bugs: Exception during shutdown

Issuing SIGTERM (Ctrl+C) to a runner server causes the following exception to be raised. 

```
023-05-01T23:27:09-0700 [ERROR] [cli] Exception in callback <bound method Arbiter.manage_watchers of <circus.arbiter.Arbiter object at 0x10719e2c0>>
Traceback (most recent call last):
  File "/opt/miniconda3/envs/bentoml/lib/python3.10/site-packages/tornado/ioloop.py", line 921, in _run
    val = self.callback()
  File "/opt/miniconda3/envs/bentoml/lib/python3.10/site-packages/circus/util.py", line 1038, in wrapper
    raise ConflictError("arbiter is already running %s command"
circus.exc.ConflictError: arbiter is already running arbiter_stop command
```

--- a/src/bentoml/_internal/marshal/dispatcher.py
+++ b/src/bentoml/_internal/marshal/dispatcher.py
@@ -234,6 +234,8 @@ class CorkDispatcher:
                     # fake wait as 0 for training requests
                     info.enqueue_time = now
                 self._loop.create_task(self.outbound_call(inputs_info))
+        except asyncio.CancelledError:
+            raise
         except Exception as e:  # pylint: disable=broad-except
             logger.error(traceback.format_exc(), exc_info=e)
 
@@ -267,6 +269,8 @@ class CorkDispatcher:
                     "BentoML has detected that a service has a max latency that is likely too low for serving. If many 503 errors are encountered, try raising the 'runner.max_latency' in your BentoML configuration YAML file."
                 )
             logger.debug("Dispatcher optimizer training complete.")
+        except asyncio.CancelledError:
+            raise
         except Exception as e:  # pylint: disable=broad-except
             logger.error(traceback.format_exc(), exc_info=e)
 

--- a/src/bentoml/_internal/marshal/dispatcher.py
+++ b/src/bentoml/_internal/marshal/dispatcher.py
@@ -234,8 +234,6 @@ class CorkDispatcher:
                     # fake wait as 0 for training requests
                     info.enqueue_time = now
                 self._loop.create_task(self.outbound_call(inputs_info))
-        except asyncio.CancelledError:
-            raise
         except Exception as e:  # pylint: disable=broad-except
             logger.error(traceback.format_exc(), exc_info=e)
 
@@ -269,8 +267,6 @@ class CorkDispatcher:
                     "BentoML has detected that a service has a max latency that is likely too low for serving. If many 503 errors are encountered, try raising the 'runner.max_latency' in your BentoML configuration YAML file."
                 )
             logger.debug("Dispatcher optimizer training complete.")
-        except asyncio.CancelledError:
-            raise
         except Exception as e:  # pylint: disable=broad-except
             logger.error(traceback.format_exc(), exc_info=e)
 

--- a/src/bentoml/_internal/marshal/dispatcher.py
+++ b/src/bentoml/_internal/marshal/dispatcher.py
@@ -235,7 +235,7 @@ class CorkDispatcher:
                     info.enqueue_time = now
                 self._loop.create_task(self.outbound_call(inputs_info))
         except asyncio.CancelledError:
-            return
+            raise
         except Exception as e:  # pylint: disable=broad-except
             logger.error(traceback.format_exc(), exc_info=e)
 

--- a/src/bentoml/_internal/marshal/dispatcher.py
+++ b/src/bentoml/_internal/marshal/dispatcher.py
@@ -313,7 +313,7 @@ class CorkDispatcher:
                 inputs_info = tuple(self._queue.pop() for _ in range(n_call_out))
                 self._loop.create_task(self.outbound_call(inputs_info))
             except asyncio.CancelledError:
-                return
+                raise
             except Exception as e:  # pylint: disable=broad-except
                 logger.error(traceback.format_exc(), exc_info=e)
 

--- a/tests/e2e/bento_server_http/tests/test_serve.py
+++ b/tests/e2e/bento_server_http/tests/test_serve.py
@@ -20,7 +20,7 @@ def test_http_server(bentoml_home: str):
 
     server.stop()
 
-    timeout = 60
+    timeout = 10
     start_time = time.time()
     while time.time() - start_time < timeout:
         retcode = server.process.poll()
@@ -48,7 +48,7 @@ def test_http_server_ctx(bentoml_home: str):
 
         assert res == {"more_test": "and more json"}
 
-    timeout = 60
+    timeout = 10
     start_time = time.time()
     while time.time() - start_time < timeout:
         retcode = server.process.poll()

